### PR TITLE
Add section about function arguments to 'Writing Documentation'

### PR DESCRIPTION
**Pull request by bordeltabernacle**

### CODE REVIEW

**Comment by bordeltabernacle** [comment]
Following on from Issue #498

I'm not sure if this is what's required.  This is as much to get some feedback on what I've done. I'm aware the example uses the `HashDict` module, which AFAIK is being deprecated, so maybe a more appropriate example is required?  Also I'm not confident in it's technical accuracy, or if this is the best, or only, way to deal with the compiler inference.  I'm also not sure if there is a *Best Practice* or any recommendations that should be included alongside this information?

File: lib/elixir/pages/Writing Documentation.md [hunk 1]
```diff
@@ -77,6 +77,20 @@ The function above can still be invoked as `MyApp.Sample.add(1, 2)`. Not only th
...
+
+When documenting a function, argument names are inferred by the compiler. This can be changed by specifying your preferred argument name.  For example:
+
+    def size(%HashDict{size: size}) do
+      size
+    end
+
+The compiler will infer this argument as `hash_dict`, whereas if you'd prefer it to be, for instance, `dict`, you can write it as so:
+
+    def size(%HashDict{size: size} = _dict) do
```
Comment (lexmag) [inline lib/elixir/pages/Writing Documentation.md]: I think it'd be better to show body-less function clause.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by bordeltabernacle** [comment]
Just a quick follow-up, `bodyless function clause` is new to me, is it documented somewhere? Thanks.

**Pull request merged**